src/ape_ethereum/ecosystem.py

```diff
@@ -467,10 +467,14 @@ def decode_calldata(self, abi: Union[ConstructorABI, MethodABI], calldata: bytes
     def decode_returndata(self, abi: MethodABI, raw_data: bytes) -> Tuple[Any, ...]:
         output_types_str_ls = [o.canonical_type for o in abi.outputs]
 
-        try:
-            vm_return_values = decode(output_types_str_ls, raw_data)
-        except (InsufficientDataBytes, NonEmptyPaddingBytes) as err:
-            raise DecodingError(str(err)) from err
+        if raw_data:
+            try:
+                vm_return_values = decode(output_types_str_ls, raw_data, strict=False)
+            except (InsufficientDataBytes, NonEmptyPaddingBytes) as err:
+                raise DecodingError(str(err)) from err
+        else:
+            # Use all zeroes.
+            vm_return_values = tuple([0 for _ in output_types_str_ls])
```

Comment:
it seems like the user is getting hardhat to give them empty bytes for a 0 return in some weird case that i havent reproduced